### PR TITLE
Merging Feature Branch

### PR DESCRIPTION
**Pull request by Kyle-Law**
- Live Version here: https://rawcdn.githack.com/Kyle-Law/library/528b0d400200b27516fec213b41da831344715cf/index.html

- User can add new book
- User can toggle read/unread
- User can remove book
- The state is persistent through local storage variable named 'myLibrary'

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+const bookPriceEl = document.getElementById('book-price');
+const bookStatusEl = document.getElementById('book-status');
+
+// Book Class
+function Book(title, author, price, status) {
+  this.title = title;
+  this.author = author;
+  this.price = price;
+  this.status = status;
+}
```
Comment (Torres-ssf) [inline script.js]: - [ ] The requirements specifically ask to implement a property `pages` for the Book object but a `price` property was implemented instead. We need to also add the `pages` property as requested.
Screenshot from 2020-08-12 11-59-41

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi There @Kyle-Law :hand: 
Awesome job on this project so far :clap: 
But we need to make some changes before moving on :+1: 

- [ ] The `.stylelintrc.json` file is different from the one provided by Microverse. We should use this [one](https://github.com/microverseinc/linters-config/blob/master/javascript/.stylelintrc.json)

- [ ] Right now is possible to create books without providing any book information. We need to block that kind of behavior and only allow to create a book when all required information is provided. We also need to be carefully to not allow the number of pages to be 0 or a negative number, like: `-50`

Please kindly submit for code review again after implementing these changes.
Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

**Comment by Kyle-Law** [comment]
Thank you for reviewing my project! Your recommendations have been implemented as below:
- Form validations are added, to prevent negative prices and pages, empty title and author
- Stylelint changed but there's a bug of unable to find csstree/validator, I check with my colleagues and got it removed. Hence that's why "csstree/validator": true is removed from .stylelintrc.json. Feel free to suggest if there's any better resolution 
- New Live Link is available [HERE](https://rawcdn.githack.com/Kyle-Law/library/58e2d078e4328eefbcb52cce7298823dadeac039/index.html)

Thanks and looking forward to another review :)

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @Kyle-Law 
### You have done well so far :clap: However, you have a few issues to fix:-
- [ ] Remove all the commented line in the codebase
- [ ] Once a user has created a book, make the modal to close after saving
- [ ] Separate app logic from the dom, this way it makes your app easy to read and maintain
- Happy Coding!

By Raymond
Slack @iamraymond

File: script.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+const bookShelf = document.getElementById('book-shelf');
+const modal = document.getElementById('modal');
+const modalShow = document.getElementById('show-modal');
+const modalClose = document.getElementById('close-modal');
+const addBookForm = document.getElementById('addbook-form');
+const bookTitleEl = document.getElementById('book-title');
+const bookAuthorEl = document.getElementById('book-author');
+const bookPagesEl = document.getElementById('book-pages');
+const bookPriceEl = document.getElementById('book-price');
+const bookStatusEl = document.getElementById('book-status');
```
Comment (rOluochKe) [inline script.js]: ### Separate this dom from logic

File: script.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+// Book Class
+function Book(title, author, pages, price, status) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.price = price;
+  this.status = status;
+}
```
Comment (rOluochKe) [inline script.js]: ### Separate this logic from dom

**Comment by Kyle-Law** [comment]
Good day! Thanks @rOluochKe for reviewing my project! Your recommendations have been implemented as followed:
- Unnecessary comments are removed
- Modal closes after a book is added
- scripts have modularized into 3 files -  Book.js, variables.js and script.js

Thanks and looking forward to another review :)

**Review by bellom — APPROVED** [review]
## Hi @Kyle-Law  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**